### ISSUE TITLE
Add drop_last setting

### ISSUE THREAD

**Issue by Raindrop182** [conservationtechlab/animl-py#330]
By default, `drop_last=False` when creating a dataloader. This means when the batch size can’t evenly divide the total amount of data, the last batch will be composed of the remaining images, which is smaller than the preset batch size.

During training, if the last batch is very small, its samples have a larger, disproportionate impact on the gradients of the model. However, most of the time, this has either a positive or negligible impact on training, as the model can overcome such slight noise. 

However, it does make an impact for EfficientNet, due to EfficientNet's use of BatchNorm layers. During evaluation, BatchNorm uses a running average accumulated during training to make calculations. This means the last uneven batch during training can skew the BatchNorm average, which impacts the validation metrics. 

The graph below displays two EfficientNet training runs in which all settings are the same, except the orange one has `drop_last=False` for the training dataloader and the gray one has `drop_last=True`

<img width="537" height="259" alt="Image" src="https://github.com/user-attachments/assets/c00a7e16-4aba-4490-a2a8-105b48229f0f" />

Therefore, I suggest adding an option in the configuration file to set `drop_last` for the train dataloader to True or False.

The reason I recommend keeping it an option rather than hardcoding `drop_last` to be True is because some exceptions exist. For instance, when the dataset is very small, you would want to set `drop_last=False` during training so that the model sees every sample possible.
